### DISCUSSION TITLE
Should I make it possible to grant control to an envoy over a full target contract?

### DISCUSSION THREAD

**Discussion by PaulRBerg** [PaulRBerg/prb-proxy#67] [Polls]
The current access-control system requires proxy owners to grant _granular_ permissions if they want to let other accounts (i.e. "envoys") execute txs on their behalf:

https://github.com/PaulRBerg/prb-proxy/blob/75944893ba4c5a6dc2b12b0ba8e33b65c19e9171/src/PRBProxy.sol#L40

However, this behavior is somewhat limiting since it asks the end user to sign lots of txs. Given that DeFi protocols typically have many moving pieces, I wonder if it wouldn't make sense to ditch the `selector` in the permission scheme so that owners can grant access to whole target contracts?

https://github.com/PaulRBerg/prb-proxy/blob/75944893ba4c5a6dc2b12b0ba8e33b65c19e9171/src/PRBProxy.sol#L192-L201

Feel free to cast your vote below. I would be particularly interested in getting a response from @razgraf, @cleanunicorn, @SolcAndMe, and @amarcu :)

**Comment by SolcAndMe** [comment]
Another approach would be to do it like this: https://github.com/fiatdao/fiat-i/blob/main/src/core/utils/Guarded.sol#L17. Though this comes with the overhead of having to check for the Wildcard selector as well.

**Reply by PaulRBerg** [reply]
Thank you for your response, @SolcAndMe.

Interesting idea with the wildcard, but I would not add more complexity to the `execute` function. The gas cost is already relatively high - higher than DSProxy, at least.

**Reply by SolcAndMe** [reply]
Yes agreed. Would also not go that route. Currently I'm in favor of just simplifying the permission scheme.

**Comment by PaulRBerg** [comment]
I ended up implementing this:

https://github.com/PaulRBerg/prb-proxy/commit/22074864458cf2bc092b705c4bb5dd37341b7fef
